File: Data Structure/Stack/LET_no739_Daily Temperatures.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        int n = temperatures.size();
        vector<int> res(n, 0);
        stack<int> s;
        
        for (int i = 0; i < n; i++) {
            int cur = temperatures[i];
            while (!s.empty() && temperatures[s.top()] < cur) {
                int j = s.top();
                s.pop();
                res[j] = i - j;
            }
            s.push(i);
        }
        
        return res;
    }
};

File: Data Structure/Stack/LET_no739_Daily Temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        n = len(temperatures)
        res = [0] * n
        stack = []
        
        for i, cur in enumerate(temperatures):
            while stack and temperatures[stack[-1]] < cur:
                j = stack.pop()
                res[j] = i - j
            stack.append(i)
        
        return res